client/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans.tsx

```diff
@@ -360,8 +364,20 @@ const useGridPlans = ( {
 						product_slug: planSlug,
 				  };
 
-		const storageAddOnsForPlan =
-			isBusinessPlan( planSlug ) || isEcommercePlan( planSlug ) ? storageAddOns : null;
+		let storageAddOnsForPlan = null;
+
+		if ( isBusinessPlan( planSlug ) || isEcommercePlan( planSlug ) ) {
+			storageAddOnsForPlan =
+				storageAddOns &&
+				storageAddOns?.map( ( addOn ) => {
+					return (
+						addOn && {
+							...addOn,
+							checkoutUrl: checkoutLink( addOn.productSlug, addOn.quantity ),
+						}
+					);
+				} );
+		}
```

Comment:
I am not sure if this transformation should be done inline here. Why not compile this in the hook that returns `storageAddOnsForPlan`?

Reply:
So for this, I think we do need to extract it into a hook. The questions I think are:

1. whether it could all be served from `useStorageAddOns` (so to include the checkout URL there), or if we need another hook `useStorageAddOnsForPlans`
2. where this new hook (`useStorageAddOnsForPlans`) should live, if we with that i.e. in the `add-ons` or `plans` data store?

Reply:
Oh my! it is too late for me 🤣 
For some reason, I got off thinking this was done inside a React component, not the `use-grid-plans` hook! So a lot less urgent to move it out of here (probably only for keeping this hook leaner)!

Reply:
> whether it could all be served from useStorageAddOns (so to include the checkout URL there)

This felt more natural. I left some of my reasoning in https://github.com/Automattic/wp-calypso/pull/83005#discussion_r1370992420 and remove the inline transformation in https://github.com/Automattic/wp-calypso/commit/5ca53a6f37850ac4389026ec1fa9d482f8d7d22b